server/game/cards/characters/05/redkeepspy.js

```diff
@@ -0,0 +1,44 @@
+const DrawCard = require('../../../drawcard.js');
+
+class RedKeepSpy extends DrawCard {
+    setupCardAbilities() {
+        this.reaction({
+            when: {
+                onCardEntersPlay: (event, card) => {
+                    return this.wasAmbush && this === card;
+                }
+            },
+            handler: () => {
+                var otherPlayer = this.game.getOtherPlayer(this.controller);
+
+                if(!otherPlayer) {
+                    return;
+                }
+
+                if(this.controller.hand.size() <= otherPlayer.hand.size()) {
```

Comment:
I would argue that this check should be part of the event, so that it doesn't trigger if this condition is not true